components/header-bar/src/profile/profile-menu.js

```diff
@@ -69,6 +69,7 @@ const ProfileContents = ({ name, email, avatarUrl, helpUrl }) => {
                             baseUrl,
                             'dhis-web-commons-security/logout.action'
                         )}
+                        onClick={onLogout}
```

Comment:
Using the `onClick` of the `MenuItem` blocks navigation, so as-is, using the `onLogout` overrides the `href` and stops navigation.

On my first review, I figured that this was intentional so would like confirmation on that. 

The way I thought this would work is that the supplier of the `onLogout` function deals with logging out, so that function would make sure that:

1. Everything is cleaned up properly
2. Trigger the logout logic
3. Redirect the browser

Reply:
@KaiVandivier actually thinking about this again... why doesn't the Headerbar logout button just call a "clearSensitiveCaches" directly from the app runtime?  I know that's another level of indirection...  Either way, this onClick needs to call the callback (if it exists) and then trigger the logout redirect

Reply:
Yeah I was unaware of the `MenuItem` implementation -- it's useful

I thought it wasn't navigating because of something with the storybook test environment 😅 

Pushed a change below

Reply:
@amcgee weird, didn't see this comment when I posted 🤔 sorry

My reasoning for using the `onLogout` callback is so that the header bar doesn't need to be 'PWA aware', and the `clearSensitiveCaches` behavior may not be generalizable to non-platform apps that are using the headerbar

Reply:
Although maybe the 'PWA aware' ship has sailed with the online status inclusion ⛵

Reply:
Actually we probably want non-platform apps to clear caches too 🤔

Reply:
Okay I think your suggestion is best -- I thought the `clearSensitiveCaches` function would live best in `@dhis2/pwa` because it's so similar to [other functions](https://github.com/dhis2/app-platform/blob/0da26e8c055122920e55791272a63dfd76ba0aa9/pwa/src/service-worker/utils.js#L64-L82) there, but I can see the app runtime would be better so `clearSCs()` can be imported and used by non-platform apps